google/stackdriver_monitoring/spectator_client.py

```diff
@@ -19,6 +19,38 @@
 import urllib2
 
 
+def __foreach_metric_tag_binding(
+    service, metric_name, metric_data,
+    visitor, visitor_pos_args, visitor_kwargs):
+  for metric_instance in metric_data['values']:
+    visitor(service, metric_name, metric_data, metric_instance,
+            *visitor_pos_args, **visitor_kwargs)
+
+
+def foreach_metric_in_service_map(
+    service_map, visitor, *visitor_pos_args, **visitor_kwargs):
+  for service, service_metrics in service_map.items():
+    for metric_name, metric_data in service_metrics.items():
+      __foreach_metric_tag_binding(
+          service, metric_name, metric_data,
+          visitor, visitor_pos_args, visitor_kwargs)
+
+
+def normalize_name_and_tags(name, metric_metadata, metric_instance):
+  tags = metric_instance.get('tags', None)
+  if not tags:
+    return name, None   # signal this metric had no tags so we can ignore it.
+
+  is_timer = metric_metadata['kind'] == 'Timer'
+  if is_timer:
+    for index, tag in enumerate(tags):
+      if tag['key'] == 'statistic':
+        name = name + '__{0}'.format(tag['value'])
+        del tags[index]
```

Comment:
It seems to work, and I'd expect it to. I would not be surprised for the iterator itself to break as you suggest, however I break immediately after doing this so never access the iterator again. Therefore, this is equivalent to breaking out of the loop and then deleting but simpler since I dont need to remember the index and carry it out of scope nor distinguish between found and unfound cases.